src/connector/src/source/filesystem/opendal_source/opendal_reader.rs

```diff
@@ -109,7 +109,7 @@ impl<Src: OpendalSource> OpendalReader<Src> {
         let object_name = split.name.clone();
 
         let reader = op
-            .reader_with(&object_name)
+            .read_with(&object_name)
```

Comment:
I don't know if this change is enough for file source, as I see its upgrade in `object_store` was a large effort (and we only upgraded that crate) https://github.com/risingwavelabs/risingwave/pull/17198.

 cc @wcy-fdu @hzxa21

Reply:
Let me try to bump opendal for connector crate.

Reply:
Another way to avoid this is to import one more opendal crate for iceberg explicitly
`iceberg_opendal = { package = "opendal", version = "0.47" }`

Reply:
I used an extra opendal for iceberg and left the comment. We can merge when update the original opendal.